packages/alveron/index.d.ts

```diff
@@ -1,35 +1,26 @@
-type Action = (state: any, ...payload: Array<any>) => [any]
-type ActionMap = {
+// exporting Value so that the consumer can import it to see what are the allowed values since TS doesn't expand types by default when viewing it in the IDE
+export type Value = string | boolean | number | { [key: string]: Value } | Array<Value>
+
+export type Model = {
```

Comment:
This only supports objects then, right?

Should also be possible to do sth. like: `const model = 0`

Reply:
Do you want to allow only object/number or do you also want strings? (like f.e. if someone wants to do `const model = "STEP_1"`)

Reply:
basically `any` here, not sure how to type it, probably looking into generics for that one soon.

Imagine a Todo App -> Array
But also want to support numbers, boolean, strings and so on.

Reply:
I guess in that case we can probably just do:
```js
export type Model = Value
```

So that we cover: string, number, boolean, array/object containing any of the previous types and also arrays/objects
That way we have pretty much an `any` scalar (excluding specific stuff like `Set`, `Map`, etc.)